### PR TITLE
Mj/fix edit person cascade

### LINKED ISSUE #160: [PE-D][Tester B] Editing person while they are in the team results in no updates
Steps to reproduce:
1. Add a person to team. `add member 1`
2. Edit persons information. `edit person 1 -n Ben`
3. The name doesn't update in the team itself

NGL I had to study how the data was saved in order to find this, good job for making it relatively bug free

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: gunbux/ped#5

### PR DESCRIPTION
**Pull request by mjgui**
Depends on #176, so go review that first...

Fixes #160

### CODE REVIEW

**Comment by marcuspang** [comment]
If this PR depends on #176, why not make that the target branch

File: src/main/java/seedu/address/model/team/TaskList.java [hunk 1]
```diff
@@ -174,8 +174,27 @@ public String toString() {
...
      */
     public void removeAssigneeIfExists(Person person) {
-        for (Task task : internalList) {
-            task.removeAssigneeIfExists(person);
+        for (int i = 0; i < internalList.size(); i++) {
+            Task task = internalList.get(i);
+            if (task.checkAssignee(person)) {
+                Task newTask = task.removeAssignee(person);
+                internalList.set(i, newTask);
+                return;
```
[the remark this answers is not in the document]
Comment (marcuspang) [reply src/main/java/seedu/address/model/team/TaskList.java:182]: This causes the update to not cascade for more than 1 task

**Review by marcuspang — CHANGES_REQUESTED** [review]
Need to remove the `return` early statements

**Comment by mjgui** [comment]
> If this PR depends on #176, why not make that the target branch

once that branch is merged into master the commits from that branch won't show up in this PR anymore

**Review by marcuspang — APPROVED** [review]
LGTM, merge conflicts tho

**Pull request merged**